src/actions/notificationsActions.js

```diff
@@ -176,9 +175,10 @@ export const startListeningNotificationsAction = () => {
         const { params: navParams = null } = getNavigationPathAndParamsState() || {};
         if (!navParams) return;
         dispatch({ type: SET_UNREAD_CHAT_NOTIFICATIONS_STATUS, payload: true });
-        if (!!navParams.username && navParams.username === notification.navigationParams.username) {
-          const contact = contacts.find(c => c.username === navParams.username) || {};
-          dispatch(getChatByContactAction(navParams.username, contact.id, contact.profileImage));
+        if (!!navParams.contact && !!navParams.contact.username
```

Comment:
Let's start using `lodash` `get` function to safely extract the data.
Then this code would look like: `if (get(navParams, ['contact', 'username'], '') === notification.navigationParams.username ... )`
or `get(navParams, 'contact.username', '')`.
You can even extract this into some var in order to re-use it

what do you think?

Reply:
Done :)